[DOCS-1209][DOCS-1204] Document full-screen panel view, various panel sharing options, customizing number of panels per section

mdlinville opened the pull request:
[DOCS-1209] Document full-screen panel view and various panel sharing options
[DOCS-1204] Customize the number of panels in a section

Previews:
https://docs-1209.docodile.pages.dev/guides/app/features/panels/#share-panels
https://docs-1209.docodile.pages.dev/guides/app/features/panels/#manage-a-sections-visible-panels

[DOCS-1209]: https://wandb.atlassian.net/browse/DOCS-1209?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DOCS-1204]: https://wandb.atlassian.net/browse/DOCS-1204?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

mdlinville committed:
[DOCS-1209] Document full-screen panel view and various panel sharing options

mdlinville committed:
[DOCS-1204] Document changing the number of panels in a section

mdlinville committed:
Fix img tag

github-actions[bot] committed:
Optimised images with calibre/image-actions

mdlinville committed:
Consolidate duplicate sections

mdlinville committed:
Merge remote-tracking branch 'origin/DOCS-1209' into DOCS-1209

connieelee commented on content/guides/models/app/features/panels/_index.md:
```diff
@@ -125,6 +151,13 @@ To add a section, click **Add section** after the last section.
+By default, each section shows 6 panels. To customize a section that has more than 6 panels:
+
+1. At the top of the section, click **1 to 6 of <X>**, where `<X>` is the total number of panels.
+1. Choose how many panels to show, or click **Show all panels**.
```
i believe we enforce a max page size of 100 panels. "show all" will only be available if num panels in section < that max

i don't think it's necessary to explicitly call out the "show all" option.

```suggestion
1. Choose how many panels to show per page.
```

mdlinville replied:
Done!

connieelee commented on content/guides/models/app/features/panels/_index.md:
```diff
@@ -61,6 +61,43 @@ To add a custom panel to your workspace:
+
+The settings of a project determine who can view the panel. This means that if the project is private, only members of the project can view the panel. If the project is public, anyone with the link can view the panel.
+
+If multiple panels have the same name, W&B shares the first panel with the name.
```
This sentence feels a little out of place here. maybe nest it under "View and share a panel in full-screen mode", or create a new section along the lines of "Construct URL by panel name".

The format of the URL is:

```
https://wandb.ai/ENTITY_NAME/PROJECT_NAME?panelDisplayName=PANEL_NAME&panelSectionName=SECTON_NAME
```

Where `PANEL_NAME` is most commonly just the metric name, and the section name param is actually optional (but is one strategy for disambiguating if there are multiple panels with the same name across the workspace)

Knowing how to construct a panel URL is probably most relevant for those creating automations (e.g. user sets up a slack bot that will alert if a metric does something weird, and they want to include a link to the panel by the metric name)

mdlinville replied:
Done!

mdlinville committed:
Connie's initial feedback

mdlinville committed:
Merge branch 'main' into DOCS-1209

connieelee commented on content/guides/models/app/features/panels/_index.md:
````diff
@@ -61,6 +61,58 @@ To add a custom panel to your workspace:
+https://wandb.ai/<ENTITY_NAME>/<PROJECT_NAME>?panelDisplayName=<PANEL_NAME>&panelSectionName=<SECTON_NAME>
+```
+
+If multiple panels have the same name, this is the URL for the first panel with the name. In this case, find the URL by [viewing the panel in full-screen mode]({{< relref "#view-and-share-a-panel-in-full-screen-mode" >}}).
````
```suggestion
If multiple panels match the specified panel and section names, the first match will be selected.
```

I hesitate to point users back to the prev section, because the URL you get from "Copy panel URL" in the UI may still have the same problem (uses panel/section name query params that have "choose first" conflict resolution)

connieelee reviewed (approved):
lgtm with minor suggestion. not sure if you need product/vis approval too?

mdlinville commented:
> lgtm with minor suggestion. not sure if you need product/vis approval too?

Who would you suggest or where should I ask for a reviewer?

connieelee commented:
> > lgtm with minor suggestion. not sure if you need product/vis approval too?
> 
> Who would you suggest or where should I ask for a reviewer?

I'm not really sure what the expected process is for docs changes either. Is an engineer the right person to approve? Does a product manager need to sign off too (in this case that would be Tristan)? Or the eng manager? @ktmathews89 I'd love to chat about this, maybe at the next team sync

ktmathews89 commented:
> > > lgtm with minor suggestion. not sure if you need product/vis approval too?
> > 
> > 
> > Who would you suggest or where should I ask for a reviewer?
> 
> I'm not really sure what the expected process is for docs changes either. Is an engineer the right person to approve? Does a product manager need to sign off too (in this case that would be Tristan)? Or the eng manager? @ktmathews89 I'd love to chat about this, maybe at the next team sync

Great question... I will add this topic to our retro doc. After we decide we can let docs team know in the DOCs ticket who should be approving.

mdlinville commented:
> > > > lgtm with minor suggestion. not sure if you need product/vis approval too?
> > > 
> > > 
> > > Who would you suggest or where should I ask for a reviewer?
> > 
> > 
> > I'm not really sure what the expected process is for docs changes either. Is an engineer the right person to approve? Does a product manager need to sign off too (in this case that would be Tristan)? Or the eng manager? @ktmathews89 I'd love to chat about this, maybe at the next team sync
> 
> Great question... I will add this topic to our retro doc. After we decide we can let docs team know in the DOCs ticket who should be approving.

Let's move internal process discussions elsewhere. :) I'll start a thread.

tristan-spaulding2 commented on content/guides/models/app/features/panels/_index.md:
```diff
@@ -61,6 +61,58 @@ To add a custom panel to your workspace:
+
+If multiple panels have the same name, this is the URL for the first panel with the name. In this case, find the URL by [viewing the panel in full-screen mode]({{< relref "#view-and-share-a-panel-in-full-screen-mode" >}}).
+
+### Share a panel with a direct link that anyone can access
```
I feel like this and the following section are basically the same thing, the difference is whether it's a public link or not.   Maybe they could be consolidated?

tristan-spaulding2 commented on content/guides/models/app/features/panels/_index.md:
```diff
@@ -61,6 +61,58 @@ To add a custom panel to your workspace:
+1. Click **Share panel in report**.
+1. In the **Share** tab, change **Only those who are invited have access** to **Anyone with the link cah view**, then choose **Share on Twitter**, **Share on Reddit**, **Share on LinkedIn**, or **Copy embed link**.
+
+### Determine a panel's full-screen URL programmatically
```
I think is hanging...you covered in a subsection above I think.

tristan-spaulding2 replied:
Maybe I'm misunderstanding.  I think it _is_ useful information we should documented, I just thought it was documented in the section "Compose a panel's full-screen URL programmatically" above.

mdlinville replied:
Yeah I think I finally understood, and have tried to address this in the latest commit.

mdlinville committed:
Merge remote-tracking branch 'origin/main' into DOCS-1209

mdlinville committed:
Connie's feedback

mdlinville committed:
Merge remote-tracking branch 'origin/DOCS-1209' into DOCS-1209

mdlinville committed:
Merge remote-tracking branch 'origin/main' into DOCS-1209

mdlinville committed:
Tristan's feedback

- Remove stray H3
- Combine the share-by-email and share-by-direct-link sections

tristan-spaulding2 commented on content/guides/models/app/features/panels/_index.md:
```diff
@@ -61,6 +61,53 @@ To add a custom panel to your workspace:
+
+When the user access the link, the panel opens in the normal view. You can instead [view and share a panel in full-screen mode]({{< relref "#view-and-share-a-panel-in-full-screen-mode" >}}).
+
+### View and share a panel in full-screen mode
```
I would like this to be the primary action we guide people toward and I worry three things here are impeding that.  

First, we're conflating the full-screen with the direct-link-to-panel points.  Yes it's great that it's full screen, but the main improvement is that you are linking someone straight to the panel, rather than (as in the section titled "### Share a panel with a direct link") creating this pretty useless one-off Report that you then share.    I think the section should first be clear on the link sharing part.   The 10000 bins thing is a detail that fits better in the docs of the full screen panel mode itself than it does here in the sharing section.

Second, I think this section should come first, before "Share a panel with a direct link", which is the action we are basically obsoleting with this feature.

Third, I think we should rename the section currently called "Share a panel with a direct link", to something like what's in the product ("Share panel in report")  so that it's not as easily confused with the new feature.    In this action, you actually are not sharing the panel with a direct link, you are secretly creating a report, and sharing the link to **that**.

mdlinville replied:
Done, PTAL!

mdlinville committed:
Merge remote-tracking branch 'origin/main' into DOCS-1209

mdlinville committed:
Tristan's feedback round 2

tristan-spaulding2 commented on content/guides/models/app/features/panels/_index.md:
```diff
@@ -61,6 +61,56 @@ To add a custom panel to your workspace:
 
+## Share a panel
+
+This section shows how to share a panel. You can also [email a snapshot of a panel]({{< relref "#email-a-snapshot-of-a-panel" >}}).
```
I would not mention this so prominently or maybe even at all.  We're trying to explain and activate them on the most convenient and powerful method which is copying the panel url

tristan-spaulding2 commented on content/guides/models/app/features/panels/_index.md:
```diff
@@ -61,6 +61,56 @@ To add a custom panel to your workspace:
+
+This section shows how to share a panel. You can also [email a snapshot of a panel]({{< relref "#email-a-snapshot-of-a-panel" >}}).
+
+The settings of a project determine who can view the panel. This means that if the project is private, only members of the project can view the panel. If the project is public, anyone with the link can view the panel.
```
This true but also in practice not very relevant since practically all customer projects are private.  Could we leave this for lower down?

mdlinville replied:
Yeah it was really needed for explaining why you might not see the social links. Let me move / rework it.

tristan-spaulding2 commented on content/guides/models/app/features/panels/_index.md:
```diff
@@ -61,6 +61,56 @@ To add a custom panel to your workspace:
+To view and share a panel in full-screen mode:
+
+1.  Complete the next steps from this menu.
+1. **Optional**: To share the panel in full-screen mode, first view the panel in full-screen mode. Hover over the panel, click the panel's action menu `...`, then click the full-screen button, which looks like a viewfinder.
```
I don't think this is true.  When you share the panel URL from the normal multi-panel workspace view, it opens in full screen.

Generally I don't know why we are documenting full screen mode and sharing in the same section, rather than documented full screen mode in its own panel actions setting unrelated to sharing.   You can full screen without sharing.

tristan-spaulding2 commented on content/guides/models/app/features/panels/_index.md:
```diff
@@ -61,6 +61,56 @@ To add a custom panel to your workspace:
+If you share a panel while viewing it in full-screen mode, the link opens the panel in full-screen mode. In full-screen mode:
+
+- A panel plots 10,000 buckets rather than 1,000 when not in full-screen mode.
+- The state of the run selector is preserved, so you can toggle runs on or off or search for runs.
```
Here's a case where documenting full screen mode and sharing together confused me at least.  I initially thought you were implying that **sharing** the link preserves the run selector state, which it does not, I think.

mdlinville committed:
Tristan's feedback round 3

mdlinville committed:
Merge branch 'main' into DOCS-1209

mdlinville committed:
Merge branch 'main' into DOCS-1209

tristan-spaulding2 commented on content/guides/models/app/features/panels/_index.md:
```diff
@@ -34,6 +34,19 @@ To reset a workspace:
+In full-screen mode:
+
+- A panel plots 10,000 buckets rather than 1,000 when not in full-screen mode.
+- The state of the run selector is preserved, so you can toggle runs on or off or search for runs.
```
I would just say "The run selector is still shown, so..."

mdlinville replied:
```suggestion
- The run selector displays, so you can toggle runs on or off and search for runs.
```

tristan-spaulding2 commented on content/guides/models/app/features/panels/_index.md:
```diff
@@ -61,6 +74,46 @@ To add a custom panel to your workspace:
+- While viewing the panel in full-screen mode, copy the URL from the browser.
+- Click the action menu `...` and select **Copy panel URL**.
+
+Share the link with the user or team. When they access the link, tpanel opens in [full-screen mode]({{< relref "#view-a-panel-in-full-screen-mode" >}}).
```
typo

mdlinville replied:
```suggestion
Share the link with the user or team. When they access the link, the panel opens in [full-screen mode]({{< relref "#view-a-panel-in-full-screen-mode" >}}).
```

tristan-spaulding2 commented on content/guides/models/app/features/panels/_index.md:
```diff
@@ -61,6 +74,46 @@ To add a custom panel to your workspace:
+
+### Email a snapshot of a panel
+
+This section shows how to email a point-in-time snapshot of a panel as a report. After you share it, the report does not update. By contrast, when you [share a panel]({{< relref "#share-a-panel" >}}).
```
The third sentence I think cuts off.  But I also don't think this is true that the plot in the report remains a point in time, rather than continuing to update as the run updates?

mdlinville replied:
PTAL!

tristan-spaulding2 commented on content/guides/models/app/features/panels/_index.md:
```diff
@@ -125,6 +167,13 @@ To add a section, click **Add section** after the last section.
 
+
+### Manage a section's visible panels
+By default, each section shows 6 panels, but you can show a maximum of 100 panels. To customize a section that has more than 6 panels:
```
I don't think this is true?  The default number depends on the width of the panels

mdlinville commented on content/guides/models/app/features/panels/_index.md:
```diff
@@ -125,6 +167,13 @@ To add a section, click **Add section** after the last section.
+By default, each section shows 6 panels, but you can show a maximum of 100 panels. To customize a section that has more than 6 panels:
+
+1. At the top of the section, click **1 to 6 of <X>**, where `<X>` is the total number of panels.
+1. Choose how many panels to show.
```
```suggestion
### Manage a section's panels
If a section has a large number of panels, by default they are paginated.

- If a section uses the **Standard grid**, you can customize the number of panels to show:
  1. At the top of the section, click **1 to <X> of <Y>**, where `<X>` is the number of visible panels and `<Y>` is the total number of panels.
  1. Choose how many panels to show, up to 100..
- To show all panels when there are a large number of them, click the section's action `...` menu, then click **Custom grid**.
- To resize a panel, hover over it and use the drag handle to resize it.
  - If the section uses the **Standard grid**, all of the section's panels are updated to the new size.
  - If the section uses the **Custom grid**, only the panel you are resizing is updated.
- To delete a panel from a section:
  1. Hover over the panel, then click its action `...` menu.
  1. Click **Delete**.
  
  If you reset a workspace to an automated workspace, all deleted panels appear again.
```

mdlinville committed:
Apply suggestions from code review

mdlinville committed:
Merge remote-tracking branch 'origin/main' into DOCS-1209

mdlinville committed:
Fix info about emailing a panel report

tristan-spaulding2 reviewed (approved):
looks good, thanks.

jtulk commented on content/guides/models/app/features/panels/_index.md:
```diff
@@ -34,6 +34,21 @@ To reset a workspace:
+To view a panel in full-screen mode:
+
+1. Hover over the panel.
+1. Click the panel's action menu `...`, then click the full-screen button, which looks like a viewfinder.
```
There's a direct full screen button you can click on each panel w/out going into the menu

mdlinville replied:
True, but then you don't get the nice "Copy panel URL" UI prompt. Instead, you must copy it from the browser bar, or go into the resulting panel's three-dots menu to copy the URL at that point. So it seems like documenting the longer way is more general-purpose, and the shorter way is discoverable while you're following the instructions for the longer way. Does that make sense?

jtulk commented on content/guides/models/app/features/panels/_index.md:
```diff
@@ -34,6 +34,21 @@ To reset a workspace:
+
+In full-screen mode:
+
+- A panel plots 10,000 buckets rather than 1,000 when not in full-screen mode.
```
I don't think this is true in sampling mode, only in Full Fidelity

mdlinville replied:
```suggestion
- In full-screen, a panel that uses full-fidelity sampling mode plots 10,000 buckets, rather than 1,000 when not in full-screen mode.
```

jtulk reviewed (commented):
I haven't worked on either sharing or managing section panels. I think workspace team is better for that.

mdlinville committed:
Update content/guides/models/app/features/panels/_index.md

mdlinville committed:
Merge branch 'main' into DOCS-1209

mdlinville committed:
Merge branch 'main' into DOCS-1209

ngrayluna commented on content/guides/models/app/features/panels/_index.md:
```diff
@@ -34,6 +34,21 @@ To reset a workspace:
+To view a panel in full-screen mode:
+
+1. Hover over the panel.
+1. Click the panel's action menu `...`, then click the full-screen button, which looks like a viewfinder.
```
Wonder if we should replace "viewfinder" with something descriptive in case users don't know what that is? Something like, "...which looks like a viewfinder icon or the outline of a square..."

mdlinville replied:
Done, PTAL

ngrayluna commented on content/guides/models/app/features/panels/_index.md:
```diff
@@ -34,6 +34,21 @@ To reset a workspace:
 
+## View a panel in full-screen mode
+
+In full-screen mode:
```
This sentence and the beginning of the first bullet point sound a little repetitive (both start with "In full-screen").  Do we need bullet points here? Suggestion to condense this to:

"In full-screen mode, the run selector displays and panels use full full-fidelity sampling mode plots with 10,000 buckets. Non full-screen mode panels use 1,000 buckets , by default.

mdlinville replied:
Done

ngrayluna commented on content/guides/models/app/features/panels/_index.md:
```diff
@@ -61,6 +76,49 @@ To add a custom panel to your workspace:
+1. Choose **Share on Twitter**, **Share on Reddit**, **Share on LinkedIn**, or **Copy embed link**.
+
+### Email a panel report
+This section shows how to email a single panel as a stand-alone report.
```
```suggestion
To email a single panel as a stand-alone report:
```

mdlinville replied:
Done

ngrayluna commented on content/guides/models/app/features/panels/_index.md:
```diff
@@ -61,6 +76,49 @@ To add a custom panel to your workspace:
+
+Share the link with the user or team. When they access the link, the panel opens in [full-screen mode]({{< relref "#view-a-panel-in-full-screen-mode" >}}).
+
+To get back to a panel's workspace while viewing a panel in full-screen mode, click the left-pointing arrow at the top of the page.
```
```suggestion
To return to a panel's workspace while viewing a panel in full-screen mode, click the left-pointing arrow at the top of the page.
```

mdlinville replied:
Done

ngrayluna commented on content/guides/models/app/features/panels/_index.md:
```diff
@@ -61,6 +76,49 @@ To add a custom panel to your workspace:
+
+This section shows how to share a panel using a link.
+
+To share a panel, you can either:
```
```suggestion
To share a panel using a link, you can either:
```

mdlinville replied:
Done

ngrayluna commented on content/guides/models/app/features/panels/_index.md:
```diff
@@ -125,6 +172,23 @@ To add a section, click **Add section** after the last section.
 
+
+### Manage a section's panels
+If a section has a large number of panels, by default they are paginated.
```
```suggestion
Sections with a large number of panels are paginated by default.
```

mdlinville replied:
Done

ngrayluna commented on content/guides/models/app/features/panels/_index.md:
```diff
@@ -125,6 +172,23 @@ To add a section, click **Add section** after the last section.
 
+
+### Manage a section's panels
+If a section has a large number of panels, by default they are paginated.
```
Can we find out how many panels and replace "large number"?

mdlinville replied:
It depends on the panel width, and each panel can be resized separately.

mdlinville replied:
Done. PTAL

ngrayluna commented on content/guides/models/app/features/panels/_index.md:
```diff
@@ -125,6 +172,23 @@ To add a section, click **Add section** after the last section.
+### Manage a section's panels
+If a section has a large number of panels, by default they are paginated.
+
+- If a section uses the **Standard grid**, you can customize the number of panels to show:
```
I'm looking at the UI and I don't see a UI element called **Standard grid** in a project workspace. Is it a UI element or a description of a type of panel layout?

If it's not a UI element, suggest replacing with italicized  and adding a definition. Same applies to Custom grid.

mdlinville replied:
Done, PTAL

mdlinville committed:
Merge remote-tracking branch 'origin/main' into DOCS-1209

mdlinville committed:
Noah's feedback

mdlinville committed:
Merge remote-tracking branch 'origin/DOCS-1209' into DOCS-1209

mdlinville committed:
Merge branch 'main' into DOCS-1209
